xscen/aggregate.py

```diff
@@ -171,34 +390,87 @@ def climatological_mean(
                 )
             ]
         else:
-            raise ValueError("The type of 'time' could not be understood.")
+            raise ValueError("The type of 'time' was not understood.")
         ds_rolling = ds_rolling.drop_vars({"month", "year", "time", "day"})
         ds_rolling = ds_rolling.assign_coords(time=time_coord).transpose("time", ...)
 
+        # append to list of results
         concats.extend([ds_rolling])
+        # end loop over periods
+
+    # concatenate results
     ds_rolling = xr.concat(concats, dim="time", data_vars="minimal")
 
-    # modify attrs and history
+    # update data_vars names, attrs, history
+    if rename_variables:
+        ds_rolling = ds_rolling.rename_vars(
+            {vv: f"{vv}_clim_{op}" for vv in ds_rolling.data_vars}
+        )
+
     for vv in ds_rolling.data_vars:
         for a in ["description", "long_name"]:
+            try:
+                op_format = dict.fromkeys(
+                    ("mean", "std", "var", "sum"), "adj"
+                ) | dict.fromkeys(("max", "min"), "noun")
+                operation = xc.core.formatting.default_formatter.format_field(
+                    op, op_format[op]
+                )
+            except (ValueError, KeyError):
+                operation = op
             update_attr(
-                ds_rolling[vv], a, _("{window}-year mean of {attr}."), window=window
+                ds_rolling[vv],
+                a,
+                _("Climatological {window}-year {operation} of {attr}."),
+                window=window,
+                operation=operation,
             )
 
         new_history = (
-            f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] {window}-year rolling average (non-centered) "
-            f"with a minimum of {min_periods} years of data - xarray v{xr.__version__}"
+            f"[{datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] Climatological {window}-year {operation} "
+            f"over window (non-centered) with a minimum of {min_periods} years of data - xarray v{xr.__version__}"
         )
         history = (
             new_history + " \n " + ds_rolling[vv].attrs["history"]
             if "history" in ds_rolling[vv].attrs
             else new_history
         )
         ds_rolling[vv].attrs["history"] = history
+
+    # update processing level
     if to_level is not None:
         ds_rolling.attrs["cat:processing_level"] = to_level
 
-    return ds_rolling
+    if periods_as_dim:
+        # restructure output to have periods as a dimension instead of stacked horizons per year/season/month
+        new_coords = {
+            1: {"year": ["ANN"]},
```

Comment:
Remind me how that ends up? Can't we use a `pandas` nomenclature?
```suggestion
            1: {"year": ["YS"]},
```

Reply:
Agreed.
Are the labels for 'season' and 'month' good as is?